### PR TITLE
Config entry tweaks

### PR DESCRIPTION
**Pull request by balloob**
Some fixes for config entries:
 - HA-Form: add support for picking an option
 - Fix overlay of paper-dialog inside app-layout for config flow wizard
 - Handle flows that finish/abort straight away

### CODE REVIEW

**Comment by andrey-git** [comment]
The alerts are just for debug, right?

**Comment by balloob** [comment]
They are not for debug, they are more a temporary stopfix until I redesign and localise everything. Previously we would just get a console error.

**Comment by balloob** [comment]
I've removed the alerts. Now if a config flow aborts/finishes right away, it will be displayed in the same form as any "abort" or "finish" message that is received by interacting with a flow.

File: panels/config/config-entries/ha-config-entries.html [hunk 1]
```diff
@@ -87,11 +88,15 @@ <h1>Add integration</h1>
...
+        _flowId: {
           type: String,
           value: null,
         },
-        userCreatedFlow: Boolean,
+        /*
+         * The step of the current selected flow, if available.
+         */
+        _flowStep: Object,
+        _userCreatedFlow: Boolean,
```
Comment (andrey-git) [inline panels/config/config-entries/ha-config-entries.html]: This is not used in bindings. Maybe make it a non-polymer property?

Comment (balloob) [reply panels/config/config-entries/ha-config-entries.html]: Fixed

File: panels/config/config-entries/ha-config-entries.html [hunk 2]
```diff
@@ -145,18 +156,20 @@ <h1>Add integration</h1>
...
-      } else if (this.userCreatedFlow) {
-        this.hass.callApi('delete', `config/config_entries/flow/${this.flowId}`);
+      } else if (this._userCreatedFlow) {
+        this.hass.callApi('delete', `config/config_entries/flow/${this._flowId}`);
       }
-      this.flowId = null;
-      this.userCreatedFlow = false;
+
+      this.setProperties({
+        _flowId: null,
```
Comment (c727) [inline panels/config/config-entries/ha-config-entries.html]: Only 1 property

Comment (balloob) [reply panels/config/config-entries/ha-config-entries.html]: fixed

**Review by andrey-git — APPROVED** [review]

**Pull request merged**